File: CPP/Arrays/11.SortZeroesnOnes.cpp
#include <bits/stdc++.h> 
using namespace std;


void sortZeronOnes(vector<int> arr){
    int start = 0;
    int end = arr.size()-1;
    int i =0;
    
    while(start <=end){
        if (arr[i] == 0){
            swap(arr[i], arr[start]);
            
            start++;
            i++;
        }
        
        if (arr[i] == 1){
            swap(arr[i], arr[end]);
            
            end--;
        }
    }
    
    for(auto val : arr){
        cout<<val<<" ";
    }
}


int main()
{
    vector<int> arr = {0, 1, 1, 0, 0, 1, 0, 1, 0, 1, 0, 1, 0};
    
    sortZeronOnes(arr);

    return 0;
}

File: Python/Arrays/11.SortZeroesnOnes.py
from array import * 

def sortZeronOne(arr):

    start = 0
    end = len(arr) -1
    i = 0

    while(start <= end):
        if arr[i] == 0:
            a = arr[i]
            arr[i] = arr[start]
            arr[start] = a

            start +=1
            i += 1

        if arr[i] == 1:
            a = arr[i]
            arr[i] = arr[end]
            arr[end] = a

            end -=1


    for i in range(len(arr)):
        print(arr[i], end=' ')

arr = array('i', [0, 1, 0, 1, 0, 1, 1, 0, 1, 0, 1, 0])

sortZeronOne(arr)